shared/actions/chat2/team-building.js

```diff
@@ -1,117 +1,109 @@
+): Promise<Array<TeamBuildingTypes.User>> =>
+  RPCTypes.apiserverGetWithSessionRpcPromise({
     args: [
-      {key: 'q', value: trimmed_query},
+      {key: 'q', value: query},
       {key: 'num_wanted', value: String(limit)},
       {key: 'service', value: service === 'keybase' ? '' : service},
-      {key: 'include_services_summary', value: '1'},
+      {key: 'include_services_summary', value: includeServicesSummary ? '1' : '0'},
     ],
     endpoint: 'user/user_search',
-  }).then(results => {
-    const users = JSON.parse(results.body)
-      .list.map(r => parseRawResultToUser(r, service))
+  }).then(results =>
+    JSON.parse(results.body)
+      .list.map(r => Constants.parseRawResultToUser(r, service))
       .filter(u => !!u)
-    return TeamBuildingGen.createSearchResultsLoaded({users, query, service})
-  })
-}
+  )
 
-const parseRawResultToUser = (
-  result: RawSearchResult,
-  service: TeamBuildingTypes.ServiceIdWithContact
-): ?TeamBuildingTypes.User => {
-  const serviceMap = Object.keys(result.services_summary).reduce((acc, service_name) => {
-    acc[service_name] = result.services_summary[service_name].username
-    return acc
-  }, {})
-
-  // Add the keybase service to the service map since it isn't there by default
-  if (result.keybase) {
-    serviceMap['keybase'] = result.keybase.username
-  }
+const searchResultCounts = (state: TypedState) => {
+  const {teamBuildingSearchQuery, teamBuildingSelectedService, teamBuildingSearchLimit} = state.chat2
 
-  if (service === 'keybase' && result.keybase) {
-    return {
-      serviceMap,
-      id: result.keybase.username,
-      prettyName: result.keybase.full_name || result.keybase.username,
-    }
-  } else if (result.service) {
-    if (result.service.service_name !== service) {
-      // This shouldn't happen
-      logger.error(
-        `Search result's service_name is different than given service name. Expected: ${service} received ${
-          result.service.service_name
-        }`
-      )
-    }
+  if (teamBuildingSearchQuery === '') {
+    return
+  }
 
-    const kbPrettyName = result.keybase && (result.keybase.full_name || result.keybase.username)
+  // filter out the service we are searching for and contact
+  const servicesToSearch = Constants.services.filter(
+    s => s !== teamBuildingSelectedService && s !== 'contact'
+  )
 
-    const prettyName =
-      result.service.full_name ||
-      kbPrettyName ||
-      `${result.service.username} on ${result.service.service_name}`
+  const isStillInSameQuery = (state: TypedState): boolean =>
+    state.chat2.teamBuildingSearchQuery === teamBuildingSearchQuery &&
+    state.chat2.teamBuildingSelectedService === teamBuildingSelectedService
+
+  return Saga.call(function*() {
+    // Defer so we aren't conflicting with the main search
+    yield Saga.call(Saga.delay, 100)
+
+    // Change this to control how many requests are in flight at a time
+    const parallelRequestsCount = 2
```

Comment:
We could, but on crappy internet it can get pretty annoying. This is nice because it can bail early if the search changes so the main search can still be fast.

Maybe if there was a way we could cancel the pending search request we could fire them all at the same time? Are we doing that anywhere else?